core-plugins/src/test/java/io/cdap/plugin/batch/joiner/JoinerConfigTest.java

```diff
@@ -182,11 +196,24 @@ public void testJoinerConfigWithDuplicateOutputFields() {
     Joiner joiner = new Joiner(config);
     Map<String, Schema> inputSchemas = ImmutableMap.of("film", filmSchema, "filmActor", filmActorSchema,
                                                        "filmCategory", filmCategorySchema);
-    joiner.init(inputSchemas);
-    joiner.getOutputSchema(inputSchemas);
+    FailureCollector collector = new MockFailureCollector();
+    String caughtException = "";
+    try {
+      joiner.init(inputSchemas, collector);
+      joiner.getOutputSchema(inputSchemas, collector);
+    } catch (Exception e) {
+      caughtException = e.getMessage();
```

Comment:
catch ValidationException and get the failures from there.

Reply:
Fixed